### ISSUE TITLE
First node of a track shouldn't have Heading. Or if it was a middle node before truncation, more could be mentioned

### ISSUE THREAD

**Issue by jidanni** [gpsbabel/gpsbabel#1019]
[We see](https://www.jidanni.org/comm/air/routes/programs/gps/tips.html) the Heading field is definitely bogus and should not be output on the first node of a trace.

But that first node could actually be fully filled out, as earlier in the same command line its heading and speed were indeed known.

Data file used:
[FlightAware_UAL871_KSFO_RCTP_20221220.kml.gz](https://github.com/GPSBabel/gpsbabel/files/10870668/FlightAware_UAL871_KSFO_RCTP_20221220.kml.gz)

**Comment by tsteven4** [comment]
This is an ancient issue with track_recompute(), which is used by the kml, gpssim and gtrnctr **writers**.  I almost wonder if this was done on purpose for some unknown reason that is hard for us to imagine now.  A fix touches 9 reference files.  While it is fair game for a writer to manipulate output data as it sees fit, it is questionable practice for a writer to change the internal route and track data as track_recompute does.  This has side effects that will be visible to any subsequent writer.  An alternative is to use the track filter course and speed options to synthesize those fields.

To fill out the your selected first node add the course option when using the track filter as shown below.  This uses a different algorithm than track_recompute with it's own peculiarities related to the initial speed and course.  However since you aren't selecting the first point the anomalies aren't visible in your output.
```
$ t=20221221032538 f=FlightAware_UAL871_KSFO_RCTP_20221220.kml
$ [gpsbabel](https://www.gpsbabel.org/) -i kml -f $f \
-x interpolate,time=1 \
-x track,course,start=$(($t-1)),stop=$t_ \
-o kml,units=a,points=1,lines=0,track=0 -F out.kml
```

**Comment by robertlipe** [comment]
That's clearly a wart. It was done in the early KML writer to give users
pretty graph data without adding a filter invocation as it built without
filters. I think gx:Track may make that less important.

It was kind of OK in the (sigh) character-converter days because the
writers were secretly given their own copy of the blob in flight because
all the data was copied to new wpt's in order to be iconv'ed and then
destroyed.  Yes, it was dueling warts that canceled each other out.

I thought I deleted GPSSim. Maybe I intended to. Maybe I just should. Do we
have any proof of users?They contracted for that years and years ago, but
they be maintaining their own local copy.

Should gtrnctr suffer that same fate?

I'm OK with a little fix that touches refs and I think  I'm OK with
track_recompute  (and 2/3rd of its callers) going away, but I've not really
thought that through.

**Comment by jidanni** [comment]
@tsteven4 Sorry to ruin your day but, no difference observed:
```
t=20221221032538 f=FlightAware_UAL871_KSFO_RCTP_20221220.kml;\
for i in "" ,course; do echo ===$i; \
gpsbabel -i kml -f $f \
 -x interpolate,time=1 -x track$i,start=$(($t-1)),stop=$(($t)) \
 -o kml,units=a,points=1,lines=0,track=0 -F -|grep -vi snippet.created|sum; done
===
35157     6
===,course
35157     6
```

**Comment by tsteven4** [comment]
Kill gpssim.
Keep gtrnctr, that is an active format, I even use it personally.

**Comment by tsteven4** [comment]
Seems good to me:
```
t=20221221032538 f=FlightAware_UAL871_KSFO_RCTP_20221220.kml;for i in "" ,course; do echo ===$i; ./gpsbabel -i kml -f $f  -x interpolate,time=1 -x track$i,start=$(($t-1)),stop=$(($t))  -o kml,units=a,points=1,lines=0,track=0 -F out.$i; done; diff out. out.,course
===
===,course
130a131
> <tr><td>Heading: 285.7</td></tr>
```

**Comment by tsteven4** [comment]
> Seems good to me:

Ahh. I had a couple changes involved.
1) don't recompute Heading if the wpt already has one.
2) don't compute Heading for the first point.

the course option takes path 1) and computes 285.7 for the first point.  Without it you take path 2) and avoid computing the erroneous 358.0.  But you don't have either change.

**Comment by robertlipe** [comment]
> Kill gpssim.
>
Done in https://github.com/GPSBabel/gpsbabel/pull/1028


> Keep gtrnctr, that is an active format, I even use it personally.
>
Certainly a customer worth keeping happy. :-)

RJL

**Comment by robertlipe** [comment]
Dan, I meant to say this before, but I'll say it more clearly now as there
are a lot of changes flying around.

Your requests are now being solve a "long time" (on a calendar) since
1.8.0. There are things that we "fixed" months ago (esp. w.r.t sub-second
handling) that haven't made it into an official release yet.  If you're not
building from source and following along, you should at least be testing
the latest builds to come out of integration after one of us has committed
a fix for something.

If you're trying to do before/after comparisons, you may need to grub
around in
https://github.com/GPSBabel/gpsbabel/actions/workflows/windows.yml
or
https://github.com/GPSBabel/gpsbabel/actions/workflows/macos.yml

Yeah, yeah. Make more releases more often. I know...

**Comment by jidanni** [comment]
No big deal, I was just making some holiday computations. I'll stick with Debian sid.

**Comment by GPSBabelDeveloper** [comment]
Debian updates from us extremely rarely.

Then they remove the update check just to be extra sure you're out of date.
:-(

**Comment by jidanni** [comment]
It seems each upstream should send a message to each maintainer at each distribution, telling them when a new version is ready.
Seems much smarter then each maintainer "being aware in their own way as to if there are new versions available."

**Comment by tsteven4** [comment]
I believe Debian monitors our repo to discover changes.  That is the purpose of https://salsa.debian.org/debian-gps-team/pkg-gpsbabel/-/blob/master/debian/watch.  Many distros do something similar.

1.8.0 has made it into sid(unstable) and bookworm(testing), but not  bullseye(stable).  I'm not sure what the hangup is.  The status is here https://tracker.debian.org/pkg/gpsbabel

**Comment by tsteven4** [comment]
Fixed in #1031
